src/lib/shim/shim_syscall.c

```diff
@@ -209,9 +219,30 @@ static void _shim_process_signals(ShimShmemHostLock* host_lock) {
             // Restore the signal-stack configuration.
             shimshmem_setSigAltStack(host_lock, shim_threadSharedMem(), ss_original);
         } else {
-            // Call signal handler with host lock released, and native syscalls disabled.
+            // It's not clear what ucontext_t would be appropriate to pass here.
+            // A signal handler that ultimately swaps to the provided the
+            // context should have arranged for the signal handler to use an
+            // alternate stack, so shouldn't get here.
+            //
+            // If we *do* ultimately want to pass a context corresponding to
+            // just before we call the handler, as in the sigaltstack case, we
+            // could construct one, perhaps with `makecontext`, or by raising
+            // and catching a native signal.
+            //
+            // If we want the context from the point at which the syscall was
+            // made, we could use the one passed to our sigsys handler for
+            // syscalls intercepted via seccomp. It's unclear what we'd do for
+            // syscalls that were intercepted via LD_PRELOAD instead, though.
+            //
+            // For now, pass NULL, and when we encounter a use-case tries to use
+            // the context it should crash or error, at which point we can
+            // revisit.
+            ucontext_t* ctx = NULL;
```

Comment:
Is it worth adding a warning message, or does this path happen often?

Reply:
Hmm, good question. I was thinking it may happen often, but right now our only real-world use case with signal handlers is the golang runtime, which so far only exercises the sigaltstack path.

Went ahead and added the warning for now. We can always reduce the severity if it's too noisy.